homeassistant/components/aurora_abb_powerone/__init__.py

```diff
@@ -47,3 +50,58 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
         hass.data[DOMAIN].pop(entry.entry_id)
 
     return unload_ok
+
+
+class AuroraAbbDataUpdateCoordinator(DataUpdateCoordinator):
+    """Class to manage fetching AuroraAbbPowerone data."""
+
+    def __init__(self, hass: HomeAssistant, comport: str, address: int) -> None:
+        """Initialize the data update coordinator."""
+        self.available_prev = False
+        self.available = False
+        self.client = AuroraSerialClient(address, comport, parity="N", timeout=1)
+        super().__init__(hass, _LOGGER, name=DOMAIN, update_interval=SCAN_INTERVAL)
+
+    def _update_data(self) -> dict[str, float]:
+        """Fetch new state data for the sensor.
+
+        This is the only function that should fetch new data for Home Assistant.
+        """
+        data: dict[str, float] = {}
+        self.available_prev = self.available
+        try:
+            self.client.connect()
+
+            # read ADC channel 3 (grid power output)
+            power_watts = self.client.measure(3, True)
+            temperature_c = self.client.measure(21)
+            energy_wh = self.client.cumulated_energy(5)
+        except AuroraTimeoutError:
+            self.available = False
+            _LOGGER.debug("No response from inverter (could be dark)")
+        except AuroraError as error:
+            self.available = False
+            raise error
+        else:
+            data["instantaneouspower"] = round(power_watts, 1)
+            data["temp"] = round(temperature_c, 1)
+            data["totalenergy"] = round(energy_wh / 1000, 2)
```

Comment:
Yes, this works as you suggest.  `suggested_display_precision` is needed otherwise you get this:
![image](https://github.com/home-assistant/core/assets/17680170/f7d22ed3-f409-41e2-890d-030d1cf739a6)
Having said that, I don't think the extra internal precision improves anything (it just takes up more bytes of storage and adds 4 lines of code).

I wondered if it would break continuity with old energy data, but it doesn't, so that's good.  If there is good reason to do this, I will add to a follow up PR.

Reply:
Oh damn, I didn't expect it to be this specific 🤣. I get why you rounded it now. Just do what you think is good :)